### ISSUE TITLE
README needed for obmc-console

### ISSUE THREAD

**Issue by vishwabmc** [openbmc/obmc-console#4]
obmc-console is using automake constructs and hence would need a README that will help whoever that makes a change and wants to recompile.

I had made some changes but could not compile it successfully with my SDK environment and I tried couple things like below and they did not compile it good. I then built a complete palmetto image and tested.
- automake
- autoreconfig
- executed the configure script

**Comment by msbarth** [comment]
There's actually more required for automake than just README.
Automake required files:
- NEWS
- README
- AUTHORS
- ChangeLog
- COPYING

I'm not sure if I've captured what was intended initially, but after you created this issue, I followed the steps that came up with on creating initial config,make, etc.. files with autotools and included them in a zip. Please re-attempt after extracting what's provided here within your repo following these steps:
- ./configure --host=arm-openbmc-linux-gnueabi
- ./config.status
- make
  [autotools_obmc-console.tar.zip](https://github.com/openbmc/obmc-console/files/371845/autotools_obmc-console.tar.zip)

**Comment by msbarth** [comment]
Looks like I missed the symlinks, here's an updated tar containing what's needed. Please update the following symlinks to point to the appropriate location within your sdk. (Currently using the default automake provided files from within the sdk)
[autotools_obmc-console.tar.zip](https://github.com/openbmc/obmc-console/files/371861/autotools_obmc-console.tar.zip)

**Comment by williamspatrick** [comment]
@jk-ozlabs - Do you know how this was set up to allow compiling?  There is no instructions.

**Comment by williamspatrick** [comment]
I was able to build on an x86 system by running just the following:

`aclocal && automake --add-missing && autoconf && ./configure`

**Comment by shenki** [comment]
I normally use the following steps to build a project from a git checkout:

```
autoreconf -i
./configure
make
```

man autoreconf:

```
NAME
       autoreconf - Update generated configuration files

SYNOPSIS
       autoreconf [OPTION]... [DIRECTORY]...

DESCRIPTION
       Run  `autoconf'  (and  `autoheader', `aclocal', `automake', `autopoint'
       (formerly `gettextize'), and `libtoolize' where appropriate) repeatedly
       to remake the GNU Build System files in specified DIRECTORIES and their
       subdirectories (defaulting to `.').
```

**Comment by williamspatrick** [comment]
@jk-ozlabs is there a recommendation?  Should we add the 'correct' way to the README or is it assumed that most people know what they are doing?

**Comment by vishwabmc** [comment]
@jk-ozlabs I would recommend for a README

**Comment by williamspatrick** [comment]
@vishwabmc - Since @msbarth is going to be converting all of the repositories over to autotools I'm not sure if it is best for us to put that in the README of each repository or it is something that is assumed to be "generally known about autotools".

**Comment by shenki** [comment]
Paging @stewart-ibm. He's our autofool

**Comment by ghost** [comment]
Generally, `autoreconf` -i`is the standard way. That being said, it's also quite common to have an`./autogen.sh`script in the repo that calls it, and this is from the days before there was just`autoreconf`and you instead had to do`autoconf; automake; aclocal` or whatever you needed.

**Comment by jk-ozlabs** [comment]
Yep, `autoreconf -i` is standard. Do we want this in all of the projects' READMEs?

**Comment by jk-ozlabs** [comment]
(In thinking that we'd have a bit of boilerplate in each readme, with the standard instructions to compile)

**Comment by williamspatrick** [comment]
@msbarth - I believe you are taking care of this by converting obmc-console to the same autotools structure we have for host-ipmid.  Please add 'Resolves openbmc/obmc-console#4.' to your commit message.

**Comment by msbarth** [comment]
Will do.

Also, from re-configuring often while converting to autotools, I planned to update the autoreconf call to 'autoreconf -if' to ensure its reconfigured each time in case autoreconf generated files already exist.
